src/main/java/com/aws/greengrass/componentmanager/plugins/DockerImageDownloader.java

```diff
@@ -188,4 +137,107 @@ public boolean canUnarchiveArtifact() {
         // Not applicable for docker images since docker engine abstracts this
         return false;
     }
+
+    private void checkDownloadPrerequisites() throws PackageDownloadException {
+        // Check that Docker engine is installed
+        if (!dockerClient.dockerInstalled()) {
+            throw new PackageDownloadException("Docker engine is not installed on the device, please ensure it's "
+                    + "installed and redo the deployment");
```

Comment:
nit: `Docker engine is not installed. Install Docker and retry the deployment.`

Reply:
Done